rasa/core/story_conflict.py

```diff
@@ -0,0 +1,252 @@
+from typing import List, Optional, Dict, Text
+
+from rasa.core.actions.action import ACTION_LISTEN_NAME
+from rasa.core.domain import PREV_PREFIX, Domain
+from rasa.core.events import ActionExecuted, Event
+from rasa.core.featurizers import MaxHistoryTrackerFeaturizer
+from rasa.nlu.constants import MESSAGE_INTENT_ATTRIBUTE
+from rasa.core.training.generator import TrackerWithCachedStates
+
+
+class StoryConflict:
+    def __init__(
+        self, sliced_states: List[Optional[Dict[Text, float]]],
+    ):
+        self.sliced_states = sliced_states
+        self.hash = hash(str(list(sliced_states)))
+        self._conflicting_actions = {}  # {"action": ["story_1", ...], ...}
+        self.correct_response = None
+
+    @staticmethod
+    def find_conflicts(
+        trackers: List[TrackerWithCachedStates], domain: Domain, max_history: int
+    ) -> List:
+        """
+        Generate a list of StoryConflict objects, describing
```

Comment:
let's please use the google docstring format as described in the coding conventions